otel_cpp/greeter_callback_client.cc

```diff
@@ -0,0 +1,137 @@
+#include "opentelemetry/sdk/metrics/meter_provider.h"
+
+#ifdef BAZEL_BUILD
+#include "otel_cpp/util.h"
+#include "protos/helloworld.grpc.pb.h"
+#else
+#include "helloworld.grpc.pb.h"
+#include "util.h"
+#endif
+
+ABSL_FLAG(std::string, target, "localhost:50051", "Server address");
+ABSL_FLAG(std::string, prometheus_endpoint, "localhost:9465",
+          "Prometheus exporter endpoint");
+
+namespace {
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+using helloworld::Greeter;
+using helloworld::HelloReply;
+using helloworld::HelloRequest;
+
+class GreeterClient {
+ public:
+  GreeterClient(std::shared_ptr<Channel> channel)
+      : stub_(Greeter::NewStub(channel)) {}
+
+  // Assembles the client's payload, sends it and presents the response back
+  // from the server.
+  std::string SayHello(const std::string& user) {
+    // Data we are sending to the server.
+    HelloRequest request;
+    request.set_name(user);
+
+    // Container for the data we expect from the server.
+    HelloReply reply;
+
+    // Context for the client. It could be used to convey extra information to
+    // the server and/or tweak certain RPC behaviors.
+    ClientContext context;
+
+    // The actual RPC.
+    std::mutex mu;
+    std::condition_variable cv;
+    bool done = false;
+    Status status;
+    stub_->async()->SayHello(&context, &request, &reply,
+                             [&mu, &cv, &done, &status](Status s) {
+                               status = std::move(s);
+                               std::lock_guard<std::mutex> lock(mu);
+                               done = true;
+                               cv.notify_one();
+                             });
+
+    std::unique_lock<std::mutex> lock(mu);
+    while (!done) {
+      cv.wait(lock);
+    }
+
+    // Act upon its status.
+    if (status.ok()) {
+      return reply.message();
+    } else {
+      std::cout << status.error_code() << ": " << status.error_message()
+                << std::endl;
+      return "RPC failed";
+    }
+  }
+
+ private:
+  std::unique_ptr<Greeter::Stub> stub_;
+};
+
+void RunClient(const std::string& target_str) {
+  // Instantiate the client. It requires a channel, out of which the actual RPCs
+  // are created. This channel models a connection to an endpoint specified by
+  // the argument "--target=" which is the only expected argument.
+  grpc::ChannelArguments args;
+  // Continuously send RPCs every second.
+  while (true) {
+    GreeterClient greeter(grpc::CreateCustomChannel(
+        target_str, grpc::InsecureChannelCredentials(), args));
+    std::string user("world");
+    std::string reply = greeter.SayHello(user);
+    std::cout << "Greeter received: " << reply << std::endl;
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+  }
+}
+
+}  // namespace
+
+int main(int argc, char** argv) {
+  absl::ParseCommandLine(argc, argv);
+
+  // CODELAB HINT : Add code to register gRPC OpenTelemetry plugin here.
```

Comment:
optional: I did something like this to make the hint more evident 

```suggestion
  ///////////////////////////////////////////////////////////////////////////
  // CODELAB HINT : Add code to register gRPC OpenTelemetry plugin here.
  ///////////////////////////////////////////////////////////////////////////
```

Reply:
done